### PR TITLE
Story/ccp 117

### PR DESCRIPTION
**Pull request by mkarthik90**
[CCP-117] Issue resolved on authentication admin user. Moved the details to property file.

### CODE REVIEW

File: Conceptpower+Spring/src/main/resources/user.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+admin=admin1234,ROLE_CP_ADMIN,enabled
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/resources/user.properties]: there needs to be at least one user coming from the pom.xml, which means the two maven variables (admin.user and admin.pw) need to be added.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java [hunk 2]
```diff
@@ -30,10 +34,21 @@
...
+	
+	
+	
 	@PostConstruct
-	public void init() {
+	public void init() throws IOException {
 		admins = new HashMap<String, String>();
+		
+		Properties prop = new Properties();
+		InputStream is = this.getClass().getResourceAsStream("/user.properties");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java]: use spring to read the properties file

Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java]: use spring to read the properties file; you can annotate classes to read specific files (qstore might have an example, or quadriga too I believe).

**Comment by jdamerow** [comment]
see comments

**Comment by mkarthik90** [comment]
@jdamerow  Resolved merge conflicts and included spring tags

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java [hunk 3]
```diff
@@ -24,17 +30,41 @@
...
+	
+	
+    @PostConstruct
+    public void init() throws IOException {
+        admins = new HashMap<String, Object>();
+        for (Iterator it = ((AbstractEnvironment) env).getPropertySources().iterator(); it.hasNext();) {
+            Object source = (Object) it.next();
+            if (source instanceof ResourcePropertySource) {
+                ResourcePropertySource propertySource = (ResourcePropertySource) source;
+                admins.putAll(((ResourcePropertySource) propertySource).getSource());
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/users/impl/UsersManager.java]: not sure if that's the best way of doing it. Why don't you call getPropertyNames() on the propertysource and iterate over the returned array of username here and add them to the admins map right away (loose the loop in line 61).

File: Conceptpower+Spring/src/main/resources/user.properties (hunk 1 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/resources/user.properties]: this is still missing

**Comment by jdamerow** [comment]
see comments

**Pull request closed without merging**